### PR TITLE
SWEET-6409 - Project skeleton

### PR DESCRIPTION
**Pull request by ddomingos-daitan**
- Have a repo with an initial skeleton of the project (app, dao, medium-tests, migrations and model) 
- Maven build passing seamlessly
- Ability to bring a local PostgreSQL instance up
- API up and running

### CODE REVIEW

File: dao/src/main/java/com/alianza/call/queue/dao/jdbi/DaoModule.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import javax.sql.DataSource;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.postgres.PostgresPlugin;
+import org.jdbi.v3.sqlobject.SqlObjectPlugin;
+
+public class DaoModule extends AbstractModule {
+
+    @Provides
```
Comment (kevinmic) [inline dao/src/main/java/com/alianza/call/queue/dao/jdbi/DaoModule.java]: This should be a singleton

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+version: '3.5'
+
+services:
+  pg:
+    image: postgres:12.1
+    environment:
+      - POSTGRES_PASSWORD=password
```
Comment (kevinmic) [inline docker-compose.yml]: Why is this here?

File: wercker.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
+no-response-timeout: 30
```
Comment (kevinmic) [inline wercker.yml]: I don't think we will be using wercker. This can probably be removed.

File: medium-tests/src/test/resources/test-config.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  loggers:
+    com.alianza.call.queue: DEBUG
+
+  appenders:
+    - type: console
+      logFormat: "%-5p [%d{ISO8601,UTC}] %c: %m mdc{%mdc}%n%rEx"
+
+database:
+  driverClass: org.postgresql.Driver
+  user: ${YOUR_JDBC_USER}
```
Comment (kevinmic) [inline medium-tests/src/test/resources/test-config.yml]: I would prefer these parameters stayed consistent with the parameters in the regular config.yml. Namely JDBC_USER instead of YOUR_JDBC_USER. Same with JDBC_PASSWORD and JDBC_URL.

File: docker-compose.yml (hunk 2 above)
Comment (kevinmic) [reply docker-compose.yml]: Specifically why are we setting POSTGRES_PASSWORD into the docker environment?

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    ports:
+      - "5432:5432"
+    volumes:
+      - "${TMPDIR:-/tmp/pg}:/var/lib/postgresql/data"
```
Comment (kevinmic) [inline docker-compose.yml]: Just curious what the "volumes" is used for?

File: app/src/test/resources/test-config.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  loggers:
+    com.alianza.call.queue: DEBUG
+
+  appenders:
+    - type: console
+      logFormat: "%-5p [%d{ISO8601,UTC}] %c: %m mdc{%mdc}%n%rEx"
+
+database:
+  driverClass: org.postgresql.Driver
+  user: ${YOUR_JDBC_USER}
```
Comment (kevinmic) [inline app/src/test/resources/test-config.yml]: Same as the comment below, remove "YOUR_" from the variable names.

File: wercker.yml (hunk 3 above)
Comment (butangero) [reply wercker.yml]: Can definitely be removed.

File: docker-compose.yml (hunk 2 above)
Comment (eberton-daitan) [reply docker-compose.yml]: > Why is this here?

You mean, an actual docker-compose file for pg?

Comment (eberton-daitan) [reply docker-compose.yml]: > Specifically why are we setting POSTGRES_PASSWORD into the docker environment?

It's just to bring up a pg instance for the local environment.

File: wercker.yml (hunk 3 above)
Comment (eberton-daitan) [reply wercker.yml]: Ok, we will remove it. Got that from call-distribution-plan.

File: medium-tests/src/test/resources/test-config.yml (hunk 4 above)
Comment (eberton-daitan) [reply medium-tests/src/test/resources/test-config.yml]: Agreed.

File: docker-compose.yml (hunk 5 above)
Comment (eberton-daitan) [reply docker-compose.yml]: Just a facilitator to really make sure you can remove the scheme from pg if necessary (by deleting the host folder).

File: dao/src/main/java/com/alianza/call/queue/dao/jdbi/DaoModule.java (hunk 1 above)
Comment (eberton-daitan) [reply dao/src/main/java/com/alianza/call/queue/dao/jdbi/DaoModule.java]: Done

File: wercker.yml (hunk 3 above)
Comment (eberton-daitan) [reply wercker.yml]: Done. Deleted the file.

File: medium-tests/src/test/resources/test-config.yml (hunk 4 above)
Comment (eberton-daitan) [reply medium-tests/src/test/resources/test-config.yml]: Diego has noticed that call-distribution-plan makes use of the **EmbeddedPostgresExtension** and also doesn't have this section and it's medium-tests is working fine so we removed it as well. If you foresee any issue with that let us know.

File: app/src/test/resources/test-config.yml (hunk 6 above)
Comment (eberton-daitan) [reply app/src/test/resources/test-config.yml]: Done

File: docker-compose.yml (hunk 5 above)
Comment (kevinmic) [reply docker-compose.yml]: Couldn't you remove pg's schema by simply recreating the postgres docker instance instead?

Comment (eberton-daitan) [reply docker-compose.yml]: Done
